File: core/src/main/scala/com/sksamuel/eel/ResultsetIterator.scala
package com.sksamuel.eel

import java.sql.ResultSet

object ResultsetIterator {
  def apply(rs: ResultSet): Iterator[Array[String]] = new Iterator[Array[String]] {
    override def hasNext: Boolean = rs.next()
    override def next(): Array[String] = {
      for ( k <- 1 to rs.getMetaData.getColumnCount ) yield rs.getString(k)
    }.toArray
  }
}

File: eel-core/src/main/kotlin/io/eels/util/ResultSetIterator.kt
package io.eels.util

import java.sql.ResultSet

class ResultSetIterator(val rs: ResultSet) : Iterator<Array<String?>> {

  override fun hasNext(): Boolean = rs.next()

  override fun next(): Array<String?> {
    val list = mutableListOf<String?>()
    for (k in 1..rs.metaData.columnCount) {
      list.add(rs.getString(k))
    }
    return list.toTypedArray()
  }
}